### PR TITLE
Add Trivia minigame

### PR DESCRIPTION
**Pull request by 1118422ILHS**
Please do a thorough review.
Intended exposed class is trivia.cs, which owns a questions.cs, which owns many AnsweredQuestions. AnsweredQuestions own AskableQuestions, which are restricted versions that don't own the knowledge of which answer is correct. Wrote three tests, including one that simulates an entire round of trivia. A fourth test that I am not willing to put in main simulated a successful trivia round, and passed.
Trivia.cs implements Minigame.cs, though it implements only one required method.

### CODE REVIEW

File: Core/WumpusCore/Minigame.cs
Comment (MaximumMaxxx) [inline Core/WumpusCore/Minigame.cs:1]: Why not make results an enum

**Comment by ILHS2000503** [comment]
git says your code failed

File: Core/WumpusCore/Trivia/AnsweredQuestion.cs
Comment (MaximumMaxxx) [inline Core/WumpusCore/Trivia/AnsweredQuestion.cs:1]: Make text a better in answerable question

**Comment by ilhs1096069** [comment]
The failed build is happening because the testing packages from nuget aren't being installed correctly. If I remember correctly we fixed this at some point, but this branch is old enough that it might not have those fixes. We might have to do a rebase?

**Comment by ilhs1096069** [comment]
Ok the issue is fixed now, github-actions needed no changes, it was the project that was broken :(

Will make a PR for this

File: Core/WumpusCore/Minigame.cs
Comment (1118422ILHS) [reply Core/WumpusCore/Minigame.cs:1]: Done.

File: Core/WumpusCore/Trivia/AnsweredQuestion.cs
Comment (1118422ILHS) [reply Core/WumpusCore/Trivia/AnsweredQuestion.cs:1]: renamed to questionText

File: Core/WumpusCore/Minigame.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+namespace WumpusCore
+{
+    public interface Minigame
+    {
+        /// <summary>
+        /// Returns whether the player won
+        /// </summary>
+        /// <returns>True if won, False if lost, and null if game not over</returns>
+        GameResult reportResult();
```
Comment (ilhs1096069) [inline Core/WumpusCore/Minigame.cs:11]: A little nit-picky but the returns comment here is a little confusing now that GameResult is an enum

**Review by MaximumMaxxx — APPROVED** [review]
I agree

**Review by ILHS2000503 — APPROVED** [review]
looks like code to me

File: Core/WumpusTesting/TriviaTesting.cs [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+using WumpusCore.Trivia;
+
+namespace WumpusTesting
+{
+    [TestClass]
+    public class TriviaTesting
+    {
+        public Trivia makeTrivia()
+        {
+            string path = Path.GetFullPath("..\\..\\..\\WumpusCore\\Trivia\\Questions.json");
```
Comment (ilhs1096069) [inline Core/WumpusTesting/TriviaTesting.cs:13]: We'll need to make sure to not use relative paths with the final game, but for testing this works

**Review by ilhs1096069 — APPROVED** [review]
Looks pretty good, one small nit-picky thing, but overall looks good to me

File: Core/WumpusTesting/TriviaTesting.cs (hunk 2 above)
Comment (1118422ILHS) [reply Core/WumpusTesting/TriviaTesting.cs:13]: its testing so its fine i think
its max's problem how to make it work in controller >:)

File: Core/WumpusCore/Minigame.cs (hunk 1 above)
Comment (1118422ILHS) [reply Core/WumpusCore/Minigame.cs:11]: Fixed

**Pull request merged**